server/websocket.go

```diff
@@ -397,6 +404,39 @@ func (p *Plugin) wsReader(us *session, handlerID string) {
 			return
 		case <-us.rtcCloseCh:
 			return
+		case <-sessionAuthTicker.C:
+			// Server versions prior to MM v9.5 won't have the session ID set so we
+			// cannot go ahead with this check.
+			// Should be removed as soon as we bump the minimum supported version.
+			if authSessionID == "" {
+				continue
+			}
+
+			if s, appErr := p.API.GetSession(authSessionID); appErr != nil || time.Now().UnixMilli() >= s.ExpiresAt {
```

Comment:
Do you have a sense of what kind of pressure this is going to put on the system? 
Is there a batched version of getSession?

Reply:
> Do you have a sense of what kind of pressure this is going to put on the system?

Performance should be fairly good since we cache sessions so normally that call shouldn't be hitting the database but yeah that's a concern I had as well. 

> Is there a batched version of getSession?

I don't believe so, definitely not exposed to the plugin side anyway.

Reply:
@cpoile To add some context, in general I felt this to be the safest approach because otherwise we'd have to implement a plugin hook that triggers on session invalidation which can definitely improve performance but it's also much easier to mess up security wise.

If performance ever becomes a problem I'd probably go down that route.